### PR TITLE
drop

### PR DESCRIPTION
**Pull request by gdementen**
see also #652

### CODE REVIEW

**Comment by gdementen** [comment]
do not review yet. I forgot to update the release notes :)

**Comment by alixdamman** [comment]
I would prefer to stay closer to pandas than numpy. So I would choose drop instead of delete.

File: larray/core/array.py [hunk 1]
```diff
@@ -5984,6 +5984,98 @@ def expand(v, length):
...
 
+    def drop(self, labels=None, indices=None, axis=None):
+        r"""Return array without some labels or indices along an axis.
+
+        Parameters
+        ----------
+        labels : scalar or Group
+            Label(s) or group to remove.
+        indices : int or list/array of ints, optional
+            Index to remove.
```
Comment (alixdamman) [inline larray/core/array.py]: s/Index/Index(ices)/?

File: larray/core/array.py [hunk 2]
```diff
@@ -5984,6 +5984,98 @@ def expand(v, length):
...
+        axis : axis reference (int, str or Axis), optional
+            Axis from which to remove labels. This is only required when using `pos` or when labels are valid on
```
Comment (alixdamman) [inline larray/core/array.py]: by `pos` you mean `indices`?

File: larray/core/array.py [hunk 3]
```diff
@@ -5984,6 +5984,98 @@ def expand(v, length):
...
         return concat(chunks, axis)
 
+    def drop(self, labels=None, indices=None, axis=None):
+        r"""Return array without some labels or indices along an axis.
+
+        Parameters
+        ----------
+        labels : scalar or Group
+            Label(s) or group to remove.
+        indices : int or list/array of ints, optional
```
Comment (alixdamman) [inline larray/core/array.py]: 1. It is not clear if it is possible to use `lables` and `indices` in the same time. I know from the implementation of the method that the answer is no. I think it should be mentioned clearly that both arguments are exclusive in the doc of the method.
2. Given the examples below, I understand that `indices` can be an IGroup right?

File: larray/core/array.py [hunk 4]
```diff
@@ -5984,6 +5984,98 @@ def expand(v, length):
...
+            Index to remove.
+        axis : axis reference (int, str or Axis), optional
+            Axis from which to remove labels. This is only required when using `pos` or when labels are valid on
+            several axes. Defaults to None, in which case the axis is derived (if present) or guessed from labels.
+
+        Returns
+        -------
+        LArray
+            Array with `labels` removed along `axis`.
```
Comment (alixdamman) [inline larray/core/array.py]: s/with `labels`/with `labels` or `indices`/

File: larray/core/array.py [hunk 5]
```diff
@@ -5984,6 +5984,98 @@ def expand(v, length):
...
+         a1   4
+
+        when deleting indices instead of labels, one must specify the axis explicitly:
+
+        >>> # arr1.drop('b.i[1]')
+        >>> arr1.drop(b.i[1])
+        a\b  b0  b2  b3
+         a0   0   2   3
+         a1   4   6   7
+
```
Comment (alixdamman) [inline larray/core/array.py]: Give also an example where you pass an int or a list of int + the axis?

**Comment by gdementen** [comment]
@alixdamman these are all good criticism... But they are also all dependent on the answer to the question I asked in #652

File: doc/source/changes/version_0_29.rst.inc [hunk 6]
```diff
@@ -1,7 +1,8 @@
 ﻿Syntax changes
 --------------
 
-* new syntax
+* LArray.insert pos and axis arguments are deprecated because those were only useful for very specific cases and those
+  can easily be rewritten by using an indices group for the "before" argument instead (closes :issue:`652`).
```
Comment (alixdamman) [inline doc/source/changes/version_0_29.rst.inc]: I'm not sure that users know what an "indices group" is. Maybe add something like (axis.i[pos]) after "an indices group".

**Comment by alixdamman** [comment]
Maybe also rename drop_labels as ignore_labels in this PR?

File: doc/source/changes/version_0_29.rst.inc (hunk 6 above)
Comment (gdementen) [reply doc/source/changes/version_0_29.rst.inc]: ok

**Pull request merged**